src/libs/Navigation/CustomActions.js

```diff
@@ -79,13 +79,20 @@ function pushDrawerRoute(route) {
             navigateBackToRootDrawer();
         }
 
-        // If we're trying to navigate to the same screen that is already active there's nothing more to do except close the drawer.
-        // This prevents unnecessary re-rendering the screen and adding duplicate items to the browser history.
         const activeState = getActiveState();
         const activeScreenName = getScreenNameFromState(activeState);
         const activeScreenParams = getParamsFromState(activeState);
-        if (newScreenName === activeScreenName && _.isEqual(activeScreenParams, newScreenParams)) {
-            return DrawerActions.closeDrawer();
+        if (newScreenName === activeScreenName && activeScreenParams.reportID === newScreenParams.reportID) {
+            // If we're trying to navigate to the same screen that is already active there's nothing more to do except close the drawer.
+            // This prevents unnecessary re-rendering the screen and adding duplicate items to the browser history.
+            if (!newScreenParams.reportActionID) {
+                return DrawerActions.closeDrawer();
+            }
+
+            // If we're trying to navigate to the same screen with a new prop then let's just set the new params and not reset the navigation.
+            if (activeScreenParams.reportActionID !== newScreenParams.reportActionID) {
+                return CommonActions.setParams(newScreenParams);
+            }
```

Comment:
Since we only set the params when they change and since we are only listening for "changes" in the component I think this is why we can't navigate to the same report action again if the params are already in the URL. So, something about this code needs to change.

Reply:
Cool, firstly its part of known limitations captured above,
> Click on a same link that was already animated before will not do anything since the URL does not change.

> I think this is why we can't navigate..

Perhaps. Right now in ReportActionsView -> componentDidUpdate i'm not scrolling if the reportActionID is unchanged. When the time comes to address this i was hoping to post a plan of action to confirm the changes and then update accordingly.